### PR TITLE
refactor(backend): properly define custom Error class for logging errors

### PR DESCRIPTION
**Pull request by ulfgebhardt**
## 🍰 Pullrequest

`LogError` class instead of `logger.error`

### Issues

- None

### Todo

- [X] None

### CODE REVIEW

File: backend/src/server/LogError.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { backendLogger as logger } from './logger'
+
+export default class LogError extends Error {
```
Comment (Mogge) [inline backend/src/server/LogError.ts:3]: Did you think about modifying the Error class itself? So we avoid those terrible imports.

**Review by Mogge — CHANGES_REQUESTED** [review]
I understand the idea, but I totally disagree with the implementation.
 * We cannot log detailed information without exposing it to the client
 * Why not override the error class itself with an optional parameter for the log?

File: backend/src/server/LogError.ts (hunk 1 above)
Comment (ulfgebhardt) [reply backend/src/server/LogError.ts:3]: I thought about it - but its advised never to do that, even tho possible. This is the proper way to do it. Further more its not required to expose anything to the client.

We can log any data without exposing it to the client.

**Comment by ulfgebhardt** [comment]
closing this in favor of https://github.com/gradido/gradido/pull/2640

**Pull request closed without merging**